cpp/src/parquet/geometry_statistics.h

```diff
@@ -0,0 +1,164 @@
+  /// (such that is_set() returns false and they should not be written).
+  EncodedGeospatialStatistics Encode() const;
+
+  /// \brief Returns true if all WKB encountered was valid or false otherwise
+  bool is_valid() const;
+
+  std::shared_ptr<GeospatialStatistics> clone() const;
+
+  /// \brief Update these statistics with previously generated statistics
+  void Decode(const EncodedGeospatialStatistics& encoded);
+
+  /// \brief The minimum encountered value in the X dimension, or Inf if no X values were
+  /// encountered.
+  ///
+  /// The Parquet definition allows for "wrap around" bounds where xmin > xmax. In this
+  /// case, these bounds represent the union of the intervals [xmax, Inf] and [-Inf,
+  /// xmin]. This implementation does not yet generate these types of bounds but they may
+  /// be encountered in files written by other readers.
+  double GetXMin() const;
+
+  /// \brief The maximum encountered value in the X dimension, or -Inf if no X values were
+  /// encountered, subject to "wrap around" bounds (see GetXMin()).
+  double GetXMax() const;
+
+  /// \brief The minimum encountered value in the Y dimension, or Inf if no Y values were
+  /// encountered.
+  ///
+  /// The Parquet definition allows for "wrap around" bounds where ymin > ymax. In this
+  /// case, these bounds represent the union of the intervals [ymax, Inf] and [-Inf,
+  /// ymin]. This implementation does not yet generate these types of bounds but they may
+  /// be encountered in files written by other readers.
+  double GetYMin() const;
+
+  /// \brief The maximum encountered value in the Y dimension, or -Inf if no Y values were
+  /// encountered, subject to "wrap around" bounds (see GetXMin()).
+  double GetYMax() const;
+
+  /// \brief The minimum encountered value in the Z dimension, or Inf if no Z values were
+  /// encountered. Wrap around bounds are not permitted in the Z dimension.
+  double GetZMin() const;
+
+  /// \brief The maximum encountered value in the Z dimension, or -Inf if no Z values were
+  /// encountered. Wrap around bounds are not permitted in the Z dimension.
+  double GetZMax() const;
+
+  /// \brief The minimum encountered value in the M dimension, or Inf if no M values were
+  /// encountered.  Wrap around bounds are not permitted in the M dimension.
+  double GetMMin() const;
+
+  /// \brief The maximum encountered value in the M dimension, or -Inf if no M values were
+  /// encountered.  Wrap around bounds are not permitted in the M dimension.
+  double GetMMax() const;
+
+  /// \brief Returns true if any Z values were encountered or false otherwise
+  bool HasZ() const;
+
+  /// \brief Returns true if any M values were encountered or false otherwise
+  bool HasM() const;
+
+  /// \brief Return the geometry type codes from the well-known binary encountered
+  ///
+  /// This implementation always returns sorted output with no duplicates.
+  std::vector<int32_t> GetGeometryTypes() const;
```

Comment:
It seems that these `GetXXX` and `HasXXX` functions are trivial and should use snake case?